src/utils/response/apiResponse.ts

```diff
@@ -1,9 +1,109 @@
+import { Translation } from '../../translations/types';
+import { UserDefinedFieldDefinitions } from '../../userDefinedFieldDefinitions/types';
+import { UserDefinedFieldDefinitionOptions } from '../../userDefinedFieldDefinitionOptions/types';
+import { Message } from '../../messages/types';
+import { Timezone } from '../../timezones/types';
+
+export type Resources = {
+  // absenceBans: AbsenceBan;
+  absenceDays: AbsenceDay;
+  // absenceReplacements: AbsenceReplacement;
+  absences: Absence;
+  absenceTypes: AbsenceType;
+  changeTimeTrackingRequests: ChangeTimeTrackingRequest;
+  // checkpoints: Checkpoint;
+  // checkpointTrackings: CheckpointTracking;
+  // checkpointTranslations: CheckpointTranslation;
+  // clients: Client;
+  teamMembers: TeamMember;
+  departments: Department;
+  deltaSync: undefined;
+  generalSettings: GeneralSetting;
+  todoTasks: TodoTask;
+  // holidayRequests: HolidayRequest;
+  messages: Message;
+  // multiuserToTasks: MultiuserToTask;
+  // nfcTransponder: NfcTransponder;
+  // nodesToUsers: NodesToUser;
+  // notifications: Notification;
+  // notificationsTypeHtml: NotificationsTypeHtml;
+  // notificationUrls: NotificationUrl;
+  // permissionResolveAbsenceTypesAndUsers: PermissionResolveAbsenceTypesAndUser;
+  // permissionResolveDepartments: PermissionResolveDepartment;
+  // permissionResolveHolidayRequests: PermissionResolveHolidayRequest;
+  // permissionResolveQuestions: PermissionResolveQuestion;
+  // permissionResolveTeams: PermissionResolveTeam;
+  // permissionResolveUsers: PermissionResolveUser;
+  // permissions: Permission;
+  recentTasks: RecentTask;
+  projects: Project;
+  timezones: Timezone;
+  // serverEvents: ServerEvent;
+  // serverTime: ServerTime;
+  // skills: Skill;
+  // surveys: Survey;
+  tasks: Task;
+  teams: Team;
+  serverCommunication: ServerCommunication;
+  timePlannings: TimePlanning;
+  timesheetAccountings: TimesheetAccounting;
+  timesheetAccountingSummaries: TimesheetAccountingSummaries;
+  // timesheetActionLogs: TimesheetActionLog;
+  // timetrackingChangelogs: TimetrackingChangelog;
+  userStatusOverviews: UserStatusOverview;
+  timeTrackings: TimeTracking;
+  translations: Translation;
+  favouriteTasks: FavouriteTask;
+  users: User;
+  usersReadMe: UserReadMe;
+  userStatusOverview: UserStatusOverview;
+  feedback: Feedback;
+  userDefinedFieldDefinitions: UserDefinedFieldDefinitions;
+  userDefinedFieldDefinitionOptions: UserDefinedFieldDefinitionOptions;
+  _EMPTY: null;
+};
+
+export type ResourceNames = keyof Resources & string;
+export type Entity<R extends ResourceNames> = Resources[R];
+
+type ListOfAllResources = { [resourceName in ResourceNames]?: Resources[resourceName][] };
+
+export type ApiResponseOnSuccess<ResourceName extends ResourceNames, Results = Resources[ResourceName][]> = {
   Success: true;
   NumResults: number;
-  Results: T;
-  RequestStartTime: string;
+  Results: Results;
+  Deleted?: ListOfAllResources;
```

Comment:
I believe Deleted is not an object, it's an array like Results but filled with something like just `id` and `deleted_at`. So the type should look similar to this: 
```  
Deleted?: {id: string, deleted_at: string}[]
```

Reply:
you are correct

Reply:
Fixed with last commit